exchangesync/node.go

```diff
@@ -0,0 +1,69 @@
+package exchangesync
+
+import (
+	"fmt"
+	"github.com/boltdb/bolt"
+	"github.com/golang/glog"
+	"github.com/open-horizon/anax/exchange"
+	"github.com/open-horizon/anax/persistence"
+	"sync"
+)
+
+var nodeUpdateLock sync.Mutex //The lock that protects the hash value
+
+var exchNode *exchange.Device
+var exchError error
+
+// Return the currently saved exchange node
+func GetExchangeNode() (*exchange.Device, error) {
+	return exchNode, exchError
+}
+
+// Return the currently saved exchange node
+func SetExchangeNode(device *exchange.Device) {
+	nodeUpdateLock.Lock()
+	defer nodeUpdateLock.Unlock()
+
+	exchNode = device
+	exchError = nil
+}
+
+// Get the node from the exchange and save it
+func SyncNodeWithExchange(db *bolt.DB, pDevice *persistence.ExchangeDevice, getDevice exchange.DeviceHandler) (*exchange.Device, error) {
+
+	glog.V(4).Infof("Checking the node changes.")
+
+	nodeUpdateLock.Lock()
+	defer nodeUpdateLock.Unlock()
+
+	// get the node user input from the exchange
+	exchDevice, err := getDevice(fmt.Sprintf("%v/%v", pDevice.Org, pDevice.Id), pDevice.Token)
```

Comment:
If you need this lock, should you really be holding it across a network call that could retry forever?

Reply:
Good catch. Fixed.